src/GameFace/Struct/UserProfile.cs

```diff
@@ -0,0 +1,148 @@
+            foreach (var record in list)
+            {
+                if (firstDayOfQuarter < record.Date && record.Date < lastDayOfQuarter)
+                {
+                    listtemp = AddElement(listtemp, new UserRecords(record.Task,1,record.Value));
+                }
+                else
+                {
+                    sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+                    listtemp = new List<UserRecords> { new UserRecords(record.Task, 1, record.Value) };
+                    date = record.Date;
+                    quarterNumber = (date.Month - 1) / 3 + 1;
+                    firstDayOfQuarter = new DateTime(date.Year, (quarterNumber - 1) * 3 + 1, 1);
+                    lastDayOfQuarter = firstDayOfQuarter.AddMonths(3).AddDays(-1);
+                }
+            }
+            if (listtemp.Count > 0)
+            {
+                sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+            }
+            return sprint;
+        }
+    }
+
+
+
+
+
+    public struct UserProfile
+    {
+        public string NickName;
+        public int Experience;
+        public List<UserAchieve> Achievements;
+        public List<UserRecords> Records;
+        public UserProfile(string nickname, int experience, List<UserAchieve> achievements, List<UserRecords> records)
+        {
+            NickName = nickname;
+            Experience = experience;
+            Achievements = achievements;
+            Records = records;
+        }
+    }
+    public struct UserHistory
+    {
+        public string NickName;
+        public int Experience;
+        public List<SprintRecords> Sprints;
+        public UserHistory(string nickname, int experience, List<SprintRecords> sprints)
+        {
+            NickName = nickname;
+            Experience = experience;
+            Sprints = sprints;
+        }
+    }
+    public struct SprintRecords
+    {
+        public DateTime StartDate;
+        public DateTime EndDate;
+        public List<UserRecords> Records;
+        public SprintRecords(DateTime startDate, DateTime endDate, List<UserRecords> records)
+        {
+            StartDate = startDate;
+            EndDate = endDate;
+            Records = records;
+        }
+    }
+
+    public struct UserRecordsDate
+    {
+        public string Task;
+        public DateTime Date;
+        public int Value;
+        public UserRecordsDate(string task, DateTime date, int value)
+        {
+            Task = task;
+            Date = date;
+            Value = value;
+        }
+    }
+    public struct UserRecords
+    {
+        public string Task { get; set; }
+        public int NoPerforms { get; set; }
+        public int Value { get; set; }
+
+        public UserRecords(string task, int noPerforms, int value)
+        {
+            Task = task;
+            NoPerforms = noPerforms;
+            Value = value;
+        }
```

Comment:
Generally speaking, it is not really necessary to have constructors that only do simple initialisation of members, like the one above.

For example, you could use object initialisers at the point of creation.
```csharp
var userRecord = new UserRecords {Task = task, NoPerforms = noPerforms, Value = value};
```

Nothing wrong either way, and the compiler treats it all the same. It is just a matter of personal preference